cmd/go-camo/main.go

```diff
@@ -180,14 +204,24 @@ func main() {
 		CamoHandler: proxy,
 	}
 
+	if opts.Metrics {
+		mlog.Printf("Enabling metrics at /metrics")
+		http.Handle("/metrics", promhttp.Handler())
+	}
+
 	if opts.Stats {
 		ps := &stats.ProxyStats{}
 		proxy.SetMetricsCollector(ps)
 		mlog.Printf("Enabling stats at /status")
 		dumbrouter.StatsHandler = stats.Handler(ps)
 	}
 
-	http.Handle("/", dumbrouter)
+	// Wrap the dumb router in instrumentation.
+	instrumentedRouter := promhttp.InstrumentHandlerDuration(responseDuration,
+		promhttp.InstrumentHandlerResponseSize(responseSize, dumbrouter),
+	)
+
+	http.Handle("/", instrumentedRouter)
```

Comment:
If metrics collection is not enabled, it might be nice to avoid the overhead of instrumentation.

Maybe something as simple as...

```
if opts.Metrics {
    instrumentedRouter := promhttp.InstrumentHandlerDuration(responseDuration,
        promhttp.InstrumentHandlerResponseSize(responseSize, dumbrouter),
    )
    http.Handle("/", instrumentedRouter)
} else {
    http.Handle("/", dumbrouter)
}
```

Reply:
I thought about that, but like I said in the issue, the overhead is on the order of 10s of CPU nanoseconds.

I thought about not even making this a flag and enabling by default. But I wanted to get your opinion on that first. From the production code I normally work with, we always enable these kinds of metrics as the overhead is extremely low.

Reply:
I think allowing it to be enabled or disabled is probably a good idea, as it fits with the existing convention for the current stats implementation. 👍